File: Reversi/Reversi/UI.cpp
//======================================
//	リバーシ UI
//======================================
#include "Mode.h"
#include "Reversi.h"
#include "Vector2.h"
#include "Utility.h"
#include <stdio.h>  // printf()

Mode SelectMode()
{
	static const char* modeName[] = {
		"１Ｐ　ＧＡＭＥ",
		"２Ｐ　ＧＡＭＥ",
		"ＷＡＴＣＨ"
	};
	int sel = 0;
	while (true) {
		ClearScreen();
		//
		// ★ここをコーディングしてください。
		//  モードを　選択して
		//  ください
		//  ＞１Ｐ　ＧＡＭＥ
		//
		//  　２Ｐ　ＧＡＭＥ
		//
		//  　ＷＡＴＣＨ
		//
		// を表示します。(カーソルは sel のところに) 
		//
		printf("モードを　選択して\nください\n");
		for (int i = 0; i < MODE_MAX; i++) {
			const char* cur = (i == sel) ? "＞": "　";
			printf("%s%s\n\n", cur, modeName[i]);
		}
		switch (GetKey()) {
		case ARROW_UP:
			sel--;
			if (sel < 0) {
				sel = MODE_MAX - 1;
			}
			break;
		case ARROW_DOWN:
			sel++;
			if (sel >= MODE_MAX) {
				sel = 0;
			}
			break;
		case DECIDE:
			return (Mode)sel;
		}
	}
}
// 位置入力
Vector2 InputPosition(Reversi* reversi)
{
	Vector2 pos = { 3,3 };
	while (true) {
		DrawScreen(reversi, pos, IN_PLAY);
		switch (GetKey()) {
			//
			// ★ここをコーディングしてください。
			// キー入力(ARROW_UP,_DOWN,_LEFT_RIGHT)によって
			//  カーソル位置(pos)を更新します
			//  pos.x は 0～BOARD_WID-1 で　右端と左端でループします。
			//  pos.y は 0～BOARD_HEI-1 で 上端と下端でループします。
			//
		case ARROW_DOWN:
			pos.y--;
			if (pos.y < 0)
				pos.y = BOARD_HEI - 1;
			break;
		case ARROW_UP:
			pos.y++;
			if (pos.y > BOARD_HEI - 1)
				pos.y = 0;
			break;
		case ARROW_LEFT:
			pos.x--;
			if (pos.x < 0)
				pos.x = BOARD_HEI - 1;
			break;
		case ARROW_RIGHT:
			pos.x++;
			if (pos.x > BOARD_HEI - 1)
				pos.x = 0;
			break;
		case DECIDE:
			if (CheckCanPlace(reversi, reversi->turn, pos) == false) {
				printf("そこには　置けません\n");
				WaitKey();
				break;
			}
			return pos;
		}
	}
}

File: Reversi_CS/Reversi_CS/UI.cs
//======================================
//      リバーシ　UI
//======================================
using System;
using Utility = GP2.Utility;
using Vector2 = GP2.Vector2;

namespace Reversi_CS
{
    internal class UI
    {
        static string[] modeName = new string[]
        {
                "１Ｐ　ＧＡＭＥ",
                "２Ｐ　ＧＡＭＥ",
                "ＷＡＴＣＨ",
        };
        // モード選択
        public static GameMode SelectMode()
        {
            int sel = 0;
            while (true)
            {
                Utility.ClearScreen();
                Utility.Puts("モードを　選択して\nください\n");
                for (int i = 0; i < modeName.Length; i++)
                {
                    string cur = (sel == i) ? "＞" : "　";
                    Utility.Printf("{0}{1}\n\n", cur, modeName[i]);
                }
                Utility.PrintOut();
                switch (Utility.GetKey())
                {
                    case ConsoleKey.UpArrow:
                        sel--;
                        if (sel < 0)
                        {
                            sel = modeName.Length - 1;
                        }
                        break;
                    case ConsoleKey.DownArrow:
                        sel++;
                        if (sel >= modeName.Length)
                        {
                            sel = 0;
                        }
                        break;
                    case ConsoleKey.Enter:
                        return (GameMode)sel;
                }
            }
        }
        // 位置入力
        public static Vector2 InputPosition(Reversi reversi)
        {
            Vector2 pos = new Vector2(3, 3);
            while (true)
            {
                reversi.DrawScreen(pos, DrawStatus.InPlay);
                switch (Utility.GetKey())
                {
                    case ConsoleKey.UpArrow:
                        pos.y--;
                        if (pos.y < 0)
                        {
                            pos.y = Reversi.BOARD_HEI - 1;
                        }
                        break;
                    case ConsoleKey.DownArrow:
                        pos.y++;
                        if (pos.y >= Reversi.BOARD_HEI)
                        {
                            pos.y = 0;
                        }
                        break;
                    case ConsoleKey.LeftArrow:
                        pos.x--;
                        if (pos.x < 0)
                        {
                            pos.x = Reversi.BOARD_WID - 1;
                        }
                        break;
                    case ConsoleKey.RightArrow:
                        pos.x++;
                        if (pos.x >= Reversi.BOARD_WID)
                        {
                            pos.x = 0;
                        }
                        break;
                    case ConsoleKey.Enter:
                        if (reversi.CheckCanPlace(reversi.turn, pos, false) == false)
                        {
                            Utility.Printf("そこには　置けません\n");
                            Utility.PrintOut();
                            Utility.WaitKey();
                            break;
                        }
                        return pos;
                }
            }
        }
    } // class
} // namespace 